subprojects/core/src/main/java/org/gradle/plugin/use/internal/InternalPluginRequest.java

```diff
@@ -17,15 +17,21 @@
 package org.gradle.plugin.use.internal;
 
 import org.gradle.api.Nullable;
-import org.gradle.plugin.internal.PluginId;
+import org.gradle.plugin.management.PluginRequest;
+import org.gradle.plugin.use.PluginId;
 
-public interface PluginRequest {
+public interface InternalPluginRequest extends PluginRequest {
```

Comment:
There is only one that is not exposed and that is `getArtifact`. That should be exposed to rules, yes.